internal/terraform/module/module_ops.go

```diff
@@ -359,19 +363,47 @@ func ParseModuleConfiguration(ctx context.Context, fs ReadOnlyFS, modStore *stat
 
 	// TODO: Avoid parsing if the content matches existing AST
 
-	// TODO: Only parse the file that's being changed/opened, unless this is 1st-time parsing
-
 	// Avoid parsing if it is already in progress or already known
 	if mod.ModuleParsingState != op.OpStateUnknown && !job.IgnoreState(ctx) {
 		return job.StateNotChangedErr{Dir: document.DirHandleFromPath(modPath)}
 	}
 
-	err = modStore.SetModuleParsingState(modPath, op.OpStateLoading)
```

Comment:
Correct, I missed adding it back. However I have to duplicate it inside the decision statement in order to correctly catch the existing StateLoaded state

Reply:
Actually, @radeksimko do we want to set the module parsing state to loading when we're only parsing a single file? Or am I reading too much into the name 'ModuleParsingState'

Reply:
This should basically only affect the condition above. Given that we always ignore the state when scheduling the job here https://github.com/hashicorp/terraform-ls/blob/94e47bd3a6371c6d56c2ab92d0d33b1ce84e9c0d/internal/indexer/document_change.go#L25 _and_ because the other branch already sets it to `Loading` here and `Loaded` here https://github.com/hashicorp/terraform-ls/blob/94e47bd3a6371c6d56c2ab92d0d33b1ce84e9c0d/internal/state/module.go#L858-L860
it shouldn't matter too much.

I think it's probably better to _not_ be setting it to `Loading` on single file parsing.